doc/tutorials/basic/progressive_layouts.md

```diff
@@ -0,0 +1,358 @@
+# Update Progressively
```

Comment:
The *Update Progressively* section is a key section to me. For me Streamlits success can to a large extend be attributed to the fact that this is easy to do for users.

Currently there is probable too much material and too little explanation. So far I've just put in what I believe *basic users* should be able to do.

## Questions

- Is this for *basic* or *intermediate* users?
- Should we only discuss one approach instead of two (generators and state/ pn.rx)? If yes then which approach?
- How can we best simplify and explain this.

For me this is a key section to get an in depth review of @philippjfr . And concrete suggestions for improvements.

Reply:
Here's what I would like the `rx` version of this code to look like:

```python
import random
from time import sleep

import panel as pn

pn.extension()

OPTIONS = ["Wind Turbine", "Solar Panel", "Battery Storage"]

def classify(image):
    sleep(2)
    return random.choice(OPTIONS)

inputs = pn.rx(list(range(0, 5)))

run = pn.widgets.Button(name="Submit", button_type="primary")

outputs = inputs.rx.for_each(
    lambda input: f"Image {input} is a {classify(input)}"
).rx.when(run, initial=["Click Submit"], during=[pn.indicators.LoadingSpinner(width=25)])

results_view = pn.Column(objects=outputs)

run.loading = outputs.rx.updating()

pn.Row(run, results_view)
```

There's some API that I've not yet added to Param though:

- `.rx.for_each` so we don't have a lambda with a list comprehension
- Ability to have `initial` and `during` placeholders on `.rx.when`